src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/AvoidStarImportCheck.java

```diff
@@ -220,8 +220,7 @@ public void visitToken(final DetailAST ast) {
             final DetailAST startingDot = ast.getFirstChild();
             logsStarredImportViolation(startingDot);
         }
-        else if (!allowStaticMemberImports
-            && ast.getType() == TokenTypes.STATIC_IMPORT) {
+        else if (!allowStaticMemberImports) {
```

Comment:
Please generate regression report with
`allowClassImports = true` and  `allowStaticMemberImports = false`

Reply:
It's done already.
The default value for both variable is false.
I have already generated a regression report for some configuration where allowClassImports was explicitly set to true, I didn't touch allowStaticMemberImports so its value was implicitly set to false.

Here is the link to the corresponding diff report: https://wilcoln.github.io/checkstyle-reports/7799/mutation/conf2/diff/index.html

Reply:
@pbludov We discussed the code change at https://github.com/checkstyle/checkstyle/issues/7799#issuecomment-599250223 and there is absolutely no way for regression to hit this. We did expand regression to be sure.